x-pack/plugins/ml/public/timeseriesexplorer/components/timeseries_chart/timeseries_chart.js

```diff
@@ -1234,6 +1408,26 @@ export class TimeseriesChart extends React.Component {
   }
 
   render() {
-    return <div className="ml-timeseries-chart-react" ref={this.setRef.bind(this)} />;
+    const { annotation, isFlyoutVisible } = this.state;
+
+    const closeFlyout = this.closeFlyout.bind(this);
+    const deleteAnnotation = this.deleteAnnotation.bind(this);
```

Comment:
We should avoid binding in render as it's expensive and impacts performance negatively because new functions are being created every time render is called and it breaks the effect of optimizations like shouldComponentUpdate. https://reactjs.org/docs/faq-functions.html#bind-in-render https://medium.freecodecamp.org/why-arrow-functions-and-bind-in-reacts-render-are-problematic-f1c08b060e36

Reply:
Good point, it was a total mix in this file with binding in constructor, class fields and binding in render 😜cleaned it up in e251767910.